GameServerCore/Domain/GameObjects/IAttackableUnit.cs

```diff
@@ -14,7 +14,8 @@ public interface IAttackableUnit : IGameObject
 
         void Die(IAttackableUnit killer);
         void TakeDamage(IAttackableUnit attacker, float damage, DamageType type, DamageSource source, bool isCrit);
-        void TakeDamage(IAttackableUnit attacker, float damage, DamageType type, DamageSource source, DamageText damageText);
+        void TakeDamage(IAttackableUnit attacker, float damage, DamageType type, DamageSource source, DamageText damageText);
+        void ApplyShield(IAttackableUnit unit, float amount, bool IsPhysical, bool IsMagical, bool StopShieldFade);
```

Comment:
```suggestion
        void ApplyShield(float amount, bool isPhysical, bool isMagical, bool noFade);
```
Also above the functions, add:
```
        float PhyShieldAmount { get; }
        float MagShieldAmount { get; }
```

Reply:
all done, already upload new version of it.